ocaml/xapi/repository_helpers.ml

```diff
@@ -622,7 +622,7 @@ let append_livepatch_guidances ~updates_info ~upd_ids_of_livepatches guidances =
     upd_ids_of_livepatches guidances
 
 let eval_guidances ~updates_info ~updates ~kind ~livepatches ~failed_livepatches
-    =
+    ~unapplied_guidances =
```

Comment:
I would suggest not to add this parameter in `eval_guidances`. The problem to be solved is to merge two guidance sets which could be considered as:
1. carry on union of two sets
2. perform `GuidanceSet.resort_guidances` on the union.

Reply:
I had a try with your suggestion: https://github.com/xapi-project/xen-api/commit/50f3c30e476f25af76e1efca9c088221683ba788
But I would prefer to keep the current version: fix by adding the `unapplied_guidances` parameter to `eval_guidances`, as otherwise there seems duplicated codes:
https://github.com/gangj/xen-api/blob/private/gangj/CA-378778.2/ocaml/xapi/repository_helpers.ml#L650-L651
https://github.com/gangj/xen-api/blob/private/gangj/CA-378778.2/ocaml/xapi/repository.ml#L775-L784
https://github.com/gangj/xen-api/blob/private/gangj/CA-378778.2/ocaml/xapi/repository.ml#L792-L801

Or could it be fixed in a better way?

Reply:
I think the duplication could be eliminated by a helper function simply:
```
let merge_guidances ~host ~kind ~guidances =
  let open GuidanceSet in
  (match kind with
  | Abosulte ->
      Db.Host.get_pending_guidances ~__context ~self:host
  | Recommended ->
      Db.Host.get_recommended_guidances ~__context ~self:host
  )
  |> List.map (fun g -> Guidance.of_update_guidance g)
  |> of_list
  |> union (of_list guidances)
  |> resort_guidances ~kind
  |> elements
```
The point is the `eval_guidances` is complicated already. And the change in this PR seems not necessary to be coded in the function.

Reply:
OK, changed by adding helper function `merge_with_unapplied_guidances`